tracer/test/Datadog.Trace.Security.IntegrationTests/AspNetCore5AsmToggle.cs

```diff
@@ -24,30 +28,37 @@ public class AspNetCore5AsmToggle : AspNetBase, IDisposable
         public AspNetCore5AsmToggle(ITestOutputHelper outputHelper)
             : base("AspNetCore5", outputHelper, "/shutdown", testName: nameof(AspNetCore5AsmToggle))
         {
+            SetEnvironmentVariable(ConfigurationKeys.Rcm.PollInterval, "500");
         }
 
         [SkippableTheory]
-        [InlineData(true)]
-        [InlineData(false)]
-        [InlineData(null)]
+        [InlineData(true, ApplyStates.ACKNOWLEDGED)]
+        [InlineData(false, ApplyStates.UNACKNOWLEDGED)]
+        [InlineData(null, ApplyStates.ACKNOWLEDGED)]
         [Trait("RunOnWindows", "True")]
-        public async Task TestSecurityToggling(bool? enableSecurity)
+        public async Task TestSecurityToggling(bool? enableSecurity, uint expectedState)
         {
-            SetEnvironmentVariable(ConfigurationKeys.Rcm.PollInterval, "500");
             var url = "/Health/?[$slice]=value";
             var agent = await RunOnSelfHosted(enableSecurity);
-            var settings = VerifyHelper.GetSpanVerifierSettings(enableSecurity);
-            var testStart = DateTime.UtcNow;
+            var settings = VerifyHelper.GetSpanVerifierSettings(enableSecurity, expectedState);
 
             var spans1 = await SendRequestsAsync(agent, url);
 
             agent.SetupRcm(Output, new[] { ((object)new Features() { Asm = new Asm() { Enabled = false } }, "1") }, "FEATURES");
-            await Task.Delay(1000);
+
+            var request1 = await agent.WaitRcmRequestAndReturnLast();
+            CheckAckState(request1, expectedState, null, "First RCM call");
+            // even the request show the applied state seems extra time is needed before it's active
+            await Task.Delay(1500);
```

Comment:
This could be executed inside `WaitRcmRequestAndReturnLast`

Reply:
It could but it's not needed in `TestRemoteConfigError()`, for example, because we don't do any further calls. 

I'm quite surprised it's needed at all, since if the client state has come through I thought that meant the config had been applied, so further waiting shouldn't be necessary. It might be interesting to find out at some point why it is necessary.